ext/gd/gd.c

```diff
@@ -1887,7 +1961,40 @@ PHP_FUNCTION(imagewebp)
 /* {{{ Output AVIF image to browser or file */
 PHP_FUNCTION(imageavif)
 {
-	_php_image_output_ctx(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_GDIMG_TYPE_AVIF, "AVIF");
+	zval *imgind;
+	zend_long quality = -1, speed = -1;
+	gdImagePtr im;
+	gdIOCtx *ctx = NULL;
+	zval *to_zval = NULL;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!ll", &imgind, gd_image_ce, &to_zval, &quality, &speed) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	im = php_gd_libgdimageptr_from_zval_p(imgind);
+
+	ctx = create_output_context(to_zval);
+	if (!ctx) {
+		RETURN_FALSE;
+	}
+
+	if (quality < -1 || quality > 100) {
+		zend_argument_value_error(3, "must be between -1 and 100");
+		ctx->gd_free(ctx);
+		RETURN_THROWS();
+	}
+	if (speed < -1 || speed > 10) {
+		zend_argument_value_error(4, "must be between -1 and 10");
+		ctx->gd_free(ctx);
+		RETURN_THROWS();
+	} else if (speed == -1) {
+		speed = 6;
+	}
```

Comment:
Ditto those checks should be before the call to `php_gd_libgdimageptr_from_zval_p`